### DISCUSSION TITLE
How to get pane object for the newly created pane

### DISCUSSION THREAD

**Discussion by eugenesvk** [wezterm/wezterm#3190] [Q&A] [answered]
I know I can get the current `pane` object from `wezterm.action_callback`
But how do I get the __newly opened__ pane object?
For example, if I map a key to two actions in `wezterm.action.Multiple`
  - `wezterm.action.SpawnTab({DomainName="local"})` to open a new tab
  - `wezterm.action_callback` to perform some actions

The `pane` object from the second callback function is the __old__ pane, not the newly opened one
Interestingly enough, when the first action is `CloseCurrentTab`, then the second callback seems to get me the correct "new" tab that WezTerm switches to, not the closed one

**Comment by wez** [comment, marked as answer]
The KeyAssignment actions are symbolic and are performed asynchronously. They do not have a return value because they have many different outcomes.

I would suggest that you use the mux api if you need to operate on the objects you're creating from inside your action callback, and just have one action callback rather than two actions. In that single callback, you would perform your actions in sequence.

https://wezfurlong.org/wezterm/config/lua/mux-window/spawn_tab.html

```lua
action_callback(function(gui_window, pane) 
   -- manually spawn a tab, rather than using SpawnTab
   local tab, pane, window = gui_window:mux_window():spawn_tab { domain = {DomainName="local"}}

   -- now you can operate on the tab/pane/window
end)
```

**Comment by eugenesvk** [comment]
Thanks! Needed to add a couple of args to replicate `SpawnCommandInNewTab` and domain `SpawnTab`, so something like this:
instead of
```lua
win:perform_action(act.SpawnCommandInNewTab(cmd_args),pane); cbEdSwitchKeyTable(win,pane)
```

I call `newTabNKey(win, pane, "CurrentPaneDomain", cmd_args)`
```lua
local function newTabNKey(win,pane, domain, cmd) -- manually spawn a tab to control its object via mux api
  local spawn_arg = {}
  if domain ~= nil then
    spawn_arg['domain'] = domain end
  if cmd    ~= nil then
    spawn_arg['args'] = cmd['args'] end
  local tab, pane, winx = win:mux_window():spawn_tab(spawn_arg)
  local win = winx:gui_window()  -- get the gui window as that's the one that has all the useful properties
  cbEdSwitchKeyTable(win,pane)  -- the original callback function that I used that now receives the newly spawned pane object
end
```

**Comment by eugenesvk** [comment]
But what is the command for the pane when I want to create a new pane split instead of a new tab? Mux window doesn't seem to have any `spawn_pane` or something
https://wezfurlong.org/wezterm/config/lua/mux-window/index.html

**Reply by wez** [reply]
You need to find the pane that you want to split, then call its [pane:split()](https://wezfurlong.org/wezterm/config/lua/pane/split.html) method

**Reply by eugenesvk** [reply]
thank you for the clarification, this work!
